systemd/systemd.go

```diff
@@ -608,10 +608,17 @@ func (s *systemd) getUnitStatus(properties []string, unitNames []string) ([]*Uni
 	cur := &UnitStatus{}
 	seen := map[string]bool{}
 
-	for _, bs := range statusregex.FindAllSubmatch(bs, -1) {
-		if len(bs[0]) == 0 {
+	// systemctl replies with a byte array which includes newlines. The regex
+	// processing the '--property=<...>' request breaks this up into single
+	// line key/value pairs, or an empty line separating properties of
+	// different units in the same request. Each line consists of an array
+	// of byte strings, i.e. [ "match", "key", "value" ].
+	for _, line := range statusregex.FindAllSubmatch(multiLineReply, -1) {
+		if len(line[0]) == 0 {
 			if len(sts) >= len(unitNames) {
-				return nil, fmt.Errorf("cannot get unit status: got more results than expected")
+				return nil, fmt.Errorf("requested: %s\ngot:\n%serror: more results than expected",
```

Comment:
this is an uncommon format for an error, as starting with "requested" doesn't make it super clear it is an error. My preference is to keep old starting:

```
cannot get unit status: got more results than expected, requested: %s
got: %s
```

or something like that

Reply:
Fixed.